tests/e2e/initialization/init.go

```diff
@@ -0,0 +1,75 @@
+package initialization
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/osmosis-labs/osmosis/v7/tests/e2e/util"
+)
+
+func InitChain(id, dataDir string, nodeConfigs []*NodeConfig, votingPeriod time.Duration, forkHeight int) (*Chain, error) {
+	chain, err := new(id, dataDir)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, nodeConfig := range nodeConfigs {
+		newNode, err := newNode(chain, nodeConfig)
+		if err != nil {
+			return nil, err
+		}
+		chain.nodes = append(chain.nodes, newNode)
+	}
+
+	if err := initGenesis(chain, votingPeriod, forkHeight); err != nil {
+		return nil, err
+	}
+
+	var peers []string
+	for _, peer := range chain.nodes {
+		peerID := fmt.Sprintf("%s@%s:26656", peer.getNodeKey().ID(), peer.moniker)
+		peer.peerId = peerID
+		peers = append(peers, peerID)
+	}
+
+	for _, node := range chain.nodes {
+		if node.isValidator {
+			if err := node.initValidatorConfigs(chain, peers); err != nil {
+				return nil, err
+			}
+		}
+	}
+	return chain.export(), nil
+}
+
+func InitSingleNode(chainId, dataDir string, existingGenesisDir string, nodeConfig *NodeConfig, votingPeriod time.Duration, trustHeight int64, trustHash string, stateSyncRPCServers []string) (*Node, error) {
+	if nodeConfig.IsValidator {
+		return nil, errors.New("creating individual validator nodes after starting up chain is not currently supported")
```

Comment:
Should we make an issue for this?

Reply:
I think we can address this when there is a need for it in the future. Currently, for testing state sync we should be able to use a regular full node instead of a validator.

I don't think there's is a current use case for us to have the validator nodes